### PR TITLE
RegisterVendor, VendorClaim and RegisterEndpoint CLI cmds execute on remote Nuts server

### LINKED ISSUE #30: one-stop-shop register organisation functionality
Change vendor claim API to support:
- generate key for new organisation when none are given
- allow registration of organisation with pri AND pub key
- add docs for administrators of vendor software (organisation registration lifecycle)

### PR DESCRIPTION
**Pull request by reinkrul**
Fixes #30

Adds remote administration for Nuts Registry. New documentation section 'Administration'.

### CODE REVIEW

File: pkg/db/memory_test.go [hunk 1]
```diff
@@ -357,7 +407,12 @@ func TestMemoryDb_ReverseLookup(t *testing.T) {
 
 func TestMemoryDb_OrganizationById(t *testing.T) {
 	t.Run("organization is found", func(t *testing.T) {
-		eventSystem, db := initDb()
+		repo, err := test.NewTestRepo(t.Name())
```
Comment (woutslakhorst) [inline pkg/db/memory_test.go]: Can some of this logic be put outside of the `t.Run` calls?

File: pkg/admin.go [hunk 2]
```diff
@@ -1,46 +1,85 @@
...
 		Status:       status,
 		Version:      version,
 	})
-	return r.EventSystem.PublishEvent(event)
 }
+
+func (r *Registry) loadOrGenerateOrganisationKey(orgID string) (map[string]interface{}, error) {
+	logrus.Infof("No keys specified for organisation (id = %s). Keys will be loaded from crypto module.", orgID)
+	entity := types.LegalEntity{URI: orgID}
+	orgKeyAsJwk, err := r.crypto.PublicKeyInJWK(entity)
```
[the remark this answers is not in the document]
Comment (reinkrul) [reply pkg/admin.go]: Made it simpler

File: engine/engine.go [hunk 3]
```diff
@@ -138,43 +143,38 @@ func cmd() *cobra.Command {
...
+			}
+			logrus.Info("Endpoint registered.")
+			logEvent(event)
 		},
 	})
 
 	return cmd
 }
+
+func logEvent(event events.Event) {
```
[the remark this answers is not in the document]
Comment (reinkrul) [reply engine/engine.go]: I think the function name should describe the intention of the function on a functional level (logging the event), how it achieves it is a matter of implementation. I'll rename it to 'logEventToConsole'

File: pkg/db/memory_test.go (hunk 1 above)
Comment (reinkrul) [reply pkg/db/memory_test.go]: Wrapped it in an initializer function

**Comment by reinkrul** [comment]
@woutslakhorst processed all comments

**Pull request merged**